When we introduce a new server we got from our uplink to our downlinks,
check the IsSendOpername() against this new server
and not on the client (thus the server link) that sent us this message.

Now the IRCd has the correct info on remote servers whether they accepted
opernames or not.

And backport n server flag propagation fix from snircd 1.4.0 - http://hg.quakenet.org/snircd/rev/711d770fb4dd

diff -r 0ea88148fd57 include/client.h
--- a/include/client.h
+++ b/include/client.h
@@ -155,6 +155,7 @@
     FLAG_HUB,                       /**< server is a hub */
     FLAG_IPV6,                      /**< server understands P10 IPv6 addrs */
     FLAG_SERVICE,                   /**< server is a service */
+    FLAG_OPERNAME,                  /**< server sends oper name in mode string */
     FLAG_GOTID,                     /**< successful ident lookup achieved */
     FLAG_DOID,                      /**< I-lines say must use ident return */
     FLAG_NONL,                      /**< No \n in buffer */
@@ -182,7 +183,6 @@
     FLAG_NOCHAN,                    /**< user's channels are hidden */
     FLAG_NOIDLE,                    /**< user's idletime is hidden */
     FLAG_XTRAOP,                    /**< oper has special powers */
-    FLAG_OPERNAME,                  /**< Server sends oper name in mode string */
 
     FLAG_LAST_FLAG,                 /**< number of flags */
     FLAG_LOCAL_UMODES = FLAG_LOCOP, /**< First local mode flag */
diff -r 0ea88148fd57 ircd/m_server.c
--- a/ircd/m_server.c
+++ b/ircd/m_server.c
@@ -775,10 +775,11 @@
       continue;
     if (0 == match(cli_name(&me), cli_name(acptr)))
       continue;
-    sendcmdto_one(sptr, CMD_SERVER, bcptr, "%s %d 0 %s %s %s%s +%s%s%s :%s",
+    sendcmdto_one(sptr, CMD_SERVER, bcptr, "%s %d 0 %s %s %s%s +%s%s%s%s :%s",
                   cli_name(acptr), hop + 1, parv[4], parv[5],
                   NumServCap(acptr), IsHub(acptr) ? "h" : "",
                   IsService(acptr) ? "s" : "", IsIPv6(acptr) ? "6" : "",
+                  IsSendOperName(acptr) ? "n" : "",
                   cli_info(acptr));
   }
   return 0;
diff -r 0ea88148fd57 ircd/s_serv.c
--- a/ircd/s_serv.c
+++ b/ircd/s_serv.c
@@ -231,7 +231,7 @@
 		    cli_hopcount(acptr) + 1, cli_serv(acptr)->timestamp,
 		    protocol_str, Protocol(acptr), NumServCap(acptr),
 		    IsHub(acptr) ? "h" : "", IsService(acptr) ? "s" : "",
-		    IsIPv6(acptr) ? "6" : "", IsSendOperName(cptr) ? "n" : "", cli_info(acptr));
+		    IsIPv6(acptr) ? "6" : "", IsSendOperName(acptr) ? "n" : "", cli_info(acptr));
     }
   }
 
diff -r 0ea88148fd57 ircd/s_stats.c
--- a/ircd/s_stats.c
+++ b/ircd/s_stats.c
@@ -541,7 +541,7 @@
                IsHub(acptr) ? 'H' : '-',
                IsService(acptr) ? 'S' : '-',
                IsIPv6(acptr) ? '6' : '-',
-               IsSendOperName(acptr) ? 'n' : '-',
+               IsSendOperName(acptr) ? 'N' : '-',
                cli_hopcount(acptr),
                NumServ(acptr),
                base64toint(cli_yxx(acptr)),
